test/e2e/nncp_cleanup_test.go

```diff
@@ -0,0 +1,53 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("NNCP cleanup", func() {
+
+	BeforeEach(func() {
+		By("Create a policy")
+		setDesiredStateWithPolicy(bridge1, linuxBrUp(bridge1))
+
+		By("Wait for policy to be ready")
+		for _, node := range nodes {
+			enactmentConditionsStatusForPolicyEventually(node, bridge1).Should(ContainElement(
+				nmstatev1alpha1.Condition{
+					Type:   nmstatev1alpha1.NodeNetworkConfigurationEnactmentConditionAvailable,
+					Status: corev1.ConditionTrue,
+				},
+			))
+		}
+	})
+
+	AfterEach(func() {
+		resetDesiredStateForNodes()
+	})
+
+	Context("when a policy is deleted", func() {
+		BeforeEach(func() {
+			deletePolicy(bridge1)
+		})
+		It("should also delete nodes enactments", func() {
+			for _, node := range nodes {
+				Eventually(func() bool {
+					key := enactmentKey(node, bridge1)
+					state := nmstatev1alpha1.NodeNetworkState{}
+					err := framework.Global.Client.Get(context.TODO(), key, &state)
+					return errors.IsNotFound(err)
+				}, 5*time.Second, 1*time.Second).Should(BeTrue())
```

Comment:
Please add a message describing what are we checking in this Should. BeTrue() is terribly vague.